api/src/test/java/org/bahmni/module/communication/service/impl/MessagingUtilityTest.java

```diff
@@ -0,0 +1,150 @@
+        messagingUtility = new MessagingUtility();
+    }
+
+    @Test
+    public void shouldReturnSessionFromMailConfigPropertiesFile() throws IOException {
+        String TMP_FOLDER = "/tmp";
+        final File file = new File(TMP_FOLDER + "/mail-config.properties");
+        FileUtils.writeStringToFile(file, "mail.send=true\n" + "mail.transport.protocol=smtp\n"
+                + "mail.smtp.host=localhost\n" + "mail.smtp.port=25\n" + "mail.smtp.auth=false\n"
+                + "mail.smtp.starttls.enable=true\n" + "mail.from=dummy@bahmni.org\n", Charset.defaultCharset());
+        OpenmrsUtil.setApplicationDataDirectory(TMP_FOLDER);
+
+        Session session = messagingUtility.getSession();
+
+        assertEquals("localhost", session.getProperty("mail.smtp.host"));
+        file.delete();
+    }
+
+    @Test
+    public void shouldReturnSameSessionIfAlreadyInitialized() throws IOException {
+        String TMP_FOLDER = "/tmp";
+        final File file = new File(TMP_FOLDER + "/mail-config.properties");
+        FileUtils.writeStringToFile(file, "mail.send=true\n" + "mail.transport.protocol=smtp\n"
+                + "mail.smtp.host=localhost\n" + "mail.smtp.port=25\n" + "mail.smtp.auth=false\n"
+                + "mail.smtp.starttls.enable=true\n" + "mail.from=dummy@bahmni.org\n", Charset.defaultCharset());
+        OpenmrsUtil.setApplicationDataDirectory(TMP_FOLDER);
+
+        Session session1 = messagingUtility.getSession();
+        Session session2 = messagingUtility.getSession();
+
+        assertSame(session1, session2);
+        file.delete();
+    }
+
+    @Test
+    public void shouldReturnSessionUsingOMRSPropertiesWhenMailConfigPropertiesFileDoesNotExist() {
+        when(administrationService.getGlobalProperty("mail.transport_protocol", "smtp")).thenReturn("");
+        when(administrationService.getGlobalProperty("mail.smtp_host", "")).thenReturn("localhost");
+        when(administrationService.getGlobalProperty("mail.smtp_port", "25")).thenReturn("25");
+        when(administrationService.getGlobalProperty("mail.smtp_auth", "false")).thenReturn("true");
+        when(administrationService.getGlobalProperty("mail.smtp.starttls.enable", "true")).thenReturn("true");
+        when(administrationService.getGlobalProperty("mail.smtp.ssl.enable", "true")).thenReturn("true");
+        when(administrationService.getGlobalProperty("mail.debug", "false")).thenReturn("false");
+        when(administrationService.getGlobalProperty("mail.from", "")).thenReturn("noreply@bahmni.org");
+        when(administrationService.getGlobalProperty("mail.user", "")).thenReturn("test_user");
+        when(administrationService.getGlobalProperty("mail.password", "")).thenReturn("test_password");
+        OpenmrsUtil.setApplicationDataDirectory("/tmp");
+
+        Session session = messagingUtility.getSession();
+
+        assertEquals("localhost", session.getProperty("mail.smtp.host"));
```

Comment:
Why are we setting so many properties in the test, and validating only one? Is it because the test fails if the mocks are not setup for all these variables?

Reply:
I'll add the rest of the assertions